File: Server/Bot/routes.Use.js

var _         = require("underscore")._
, Bot_Use       = require("./model").Bot_Use
, Screen_Name = require("../Screen_Name/model").Screen_Name
, Website     = require("../Website/model").Website
, log         = require("../App/base").log
;

exports.route = function (mod) {
  var app = mod.app;

  // ============ CREATE ===============================================

  app.post("/Bot_Use", function (req, resp, next) {
    var data = _.clone(req.body);
    mod.New_River(req, resp, next)
    .read_one('screen_name', function (j) {
      Screen_Name.read_by_screen_name(req.params.screen_name, req.user, j);
    })
    .create_one(function (j, sn) {
      Bot_Use.create_by_screen_name(sn, data, j);
    })
    .job(function (j, model) {
      resp.json({
        success : true,
        msg     : 'Created: ',
        model   : model.public_data()
      });
    })
    .run();
  });

  // ============ READ =================================================

  app.get("/Bot_Use/:id", function (req, resp, next) {
    var OK            = mod.New_Request(arguments);
    var opts          = OK.template_data('Bot_Use/show_one');
    opts['title']     = "Bot_Use #" + req.params.id;
    return OK.render_template();
  });

}; // ==== exports.routes

File: Server/Bot/routes.Use.rb

# ============ CREATE ===============================================

post '/Bot/Use' do

  begin
    o = Bot::Use.create({
      bot: params[:bot],
      owner: params[:as_this_life]
    })

    json true, "You are now using, #{o.screen_name}, with #{use.owner}."
  rescue Bot::Use::Invalid => e
    json false, e.msg
  end

end # === post /Bot/Use


post "/Bot_Use" do
  sn = Screen_Name.read_by_screen_name(params[:screen_name], request.user)
  o  = Bot_Use.create_by_screen_name(sn, params)
  json true,'Created: ', model: o.to_public
end

# ============ READ =================================================

get "/Bot_Use/:id" do
  render 'Bot_Use/show_one', :title=> "Bot_Use ##{req.params.id}"
end



